src/WebJobs.Extensions.DocumentDB/DocumentDBContext.cs

```diff
@@ -1,13 +1,24 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using Microsoft.Azure.WebJobs.Host;
+
 namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
 {
     internal class DocumentDBContext
     {
+        private const int DefaultMaxThrottleRetries = 10;
+
+        public DocumentDBContext()
+        {
+            MaxThrottleRetries = DefaultMaxThrottleRetries;
+        }
+
         public IDocumentDBService Service { get; set; }
         public string ResolvedDatabaseName { get; set; }
         public string ResolvedCollectionName { get; set; }
         public string ResolvedId { get; set; }
+        public int MaxThrottleRetries { get; set; }
+        public TraceWriter Trace { get; set; }
```

Comment:
Even though I'm doing no tracing anywhere right now, I decided to leave this property here. It should make debugging easier in the future.